src/model/voice.rs

```diff
@@ -49,13 +58,229 @@ impl fmt::Debug for VoiceState {
+                    if self_mute.is_some() {
+                        return Err(de::Error::duplicate_field("self_mute"));
+                    }
+                    self_mute = Some(map.next_value()?);
+                }
+                Field::SelfStream => {
+                    if self_stream.is_some() {
+                        return Err(de::Error::duplicate_field("self_stream"));
+                    }
+                    self_stream = map.next_value()?;
+                }
+                Field::SelfVideo => {
+                    if self_video.is_some() {
+                        return Err(de::Error::duplicate_field("self_video"));
+                    }
+                    self_video = Some(map.next_value()?);
+                }
+                Field::SessionId => {
+                    if session_id.is_some() {
+                        return Err(de::Error::duplicate_field("session_id"));
+                    }
+                    session_id = Some(map.next_value()?);
+                }
+                Field::Suppress => {
+                    if suppress.is_some() {
+                        return Err(de::Error::duplicate_field("suppress"));
+                    }
+                    suppress = Some(map.next_value()?);
+                }
+                Field::Token => {
+                    if token.is_some() {
+                        return Err(de::Error::duplicate_field("token"));
+                    }
+                    token = map.next_value()?;
+                }
+                Field::UserId => {
+                    if user_id.is_some() {
+                        return Err(de::Error::duplicate_field("user_id"));
+                    }
+                    user_id = Some(map.next_value()?);
+                }
+            }
+        }
+
+        let deaf = deaf.ok_or_else(|| de::Error::missing_field("deaf"))?;
+        let mute = mute.ok_or_else(|| de::Error::missing_field("mute"))?;
+        let self_deaf = self_deaf.ok_or_else(|| de::Error::missing_field("self_deaf"))?;
+        let self_mute = self_mute.ok_or_else(|| de::Error::missing_field("self_mute"))?;
+        let self_video = self_video.ok_or_else(|| de::Error::missing_field("self_video"))?;
+        let session_id = session_id.ok_or_else(|| de::Error::missing_field("session_id"))?;
+        let suppress = suppress.ok_or_else(|| de::Error::missing_field("suppress"))?;
+        let user_id = user_id.ok_or_else(|| de::Error::missing_field("user_id"))?;
+
+        if let (Some(guild_id), Some( member)) = (guild_id, member.as_mut()) {
+            member.guild_id = guild_id;
+        }
+
+        Ok(VoiceState {
+            channel_id,
+            deaf,
+            guild_id,
+            member,
+            mute,
+            self_deaf,
+            self_mute,
+            self_stream,
+            self_video,
+            session_id,
+            suppress,
+            token,
+            user_id,
+            _nonexhaustive: ()
+        })
+    }
+}
```

Comment:
These should be placed inside the `deserialize` function below. Outside of deserialization of `VoiceState`, these types are useless and pollute the module's namespace.